raiden/network/transport/matrix/rtc/web_rtc.py

```diff
@@ -0,0 +1,216 @@
+    await pc.setLocalDescription(offer)
+
+    log.debug("Created offer", offer=offer)
+
+    msg = "Channel must be created already"
+    assert rtc_partner.channel is not None, msg
+
+    @rtc_partner.channel.on("message")
+    async def on_message(message: Dict[str, bytes]) -> None:  # pylint: disable=unused-variable
+        log.debug(
+            "Received message in asyncio kingdom",
+            node=to_checksum_address(node_address),
+            message=message,
+            time=time.time(),
+        )
+        handle_message_callback(message, rtc_partner.partner_address)
+
+    @rtc_partner.channel.on("close")
+    def on_close() -> None:  # pylint: disable=unused-variable
+        rtc_partner.channel = None
+
+    return pc.localDescription
+
+
+async def set_remote_description(
+    rtc_partner: RTCPartner,
+    node_address: Address,
+    description: Dict[str, str],
+    handle_message_callback: Callable,
+) -> Optional[RTCSessionDescription]:
+
+    remote_description = RTCSessionDescription(description["sdp"], description["type"])
+    sdp_type = description["type"]
+    pc = rtc_partner.peer_connection
+    await pc.setRemoteDescription(remote_description)
+
+    @rtc_partner.peer_connection.on("datachannel")
+    def on_datachannel(channel: RTCDataChannel) -> None:  # pylint: disable=unused-variable
+        rtc_partner.channel = channel
+        log.debug(f"Received rtc channel {channel.label}", node=to_checksum_address(node_address))
+
+        @channel.on("close")
+        def channel_closed() -> None:  # pylint: disable=unused-variable
+            rtc_partner.channel = None
+
+        @channel.on("message")
+        async def on_message(message: Dict[str, bytes]) -> None:  # pylint:disable=unused-variable
+            log.debug(
+                "Received message in asyncio kingdom",
+                node=to_checksum_address(node_address),
+                message=message,
+                time=time.time(),
+            )
+            handle_message_callback(message, rtc_partner.partner_address)
+
+        @channel.on("close")
+        async def on_close() -> None:  # pylint: disable=unused-variable
+            rtc_partner.channel = None
+
+    if sdp_type == SDPTypes.OFFER.value:
+        # send answer
+        answer = await pc.createAnswer()
+        await pc.setLocalDescription(answer)
+
+        return pc.localDescription
+    return None
+
+
+def send_message(rtc_partner: RTCPartner, message: str, node_address: Address) -> None:
+    channel = rtc_partner.channel
+    if channel is not None and channel.readyState == "open":
+        log.debug(
+            "Sending message in asyncio kingdom",
+            node=to_checksum_address(node_address),
+            message=message,
+            time=time.time(),
+        )
+        channel.send(message)
+    else:
+        log.debug(
+            "Channel is not open. ReadyState: "
+            f"{channel.readyState if channel is not None else 'No channel exists'}",
```

Comment:
Nitpick: Unnecessary f-string is unnecessary ;)
This could also just be a kwarg instead being baked into the event.